---
title: "Intro to Epigenomics Claire He"
author: Wade Boohar
date: 11/03/24
updated: 03/07/24
---


```{r setup}
 knitr::opts_knit$set(root.dir = normalizePath("/home1/heclaire/490_cluster/analysis_data")) 
```


Package Download and Data-cleaning
```{r}
if (!require("sesameData", quietly = TRUE))
BiocManager::install("sesameData")

if (!require("sesame", quietly = TRUE))
BiocManager::install("sesame")

if (!require("limma", quietly = TRUE))
BiocManager::install("limma")
```


Load in all necessary packages
```{r}
library(TCGAbiolinks)
library(sesame)
library(sesameData)
library(limma)
library(ggplot2)
library(BiocManager)
library(maftools)
library(DESeq2)
```

```{r}
query <- GDCquery(project = "TCGA-BRCA",
                  data.category = "DNA Methylation",
                  data.type = "Methylation Beta Value",
                  platform = "Illumina Human Methylation 450",
                  data.format = "Dataframe")
GDCdownload(query)
methylation450 <- GDCprepare(query)
```

```{r}
methylation_clinical <- as.data.frame(methylation450@colData)
betas <- as.data.frame(methylation450@assays@data@listData)
cpg_sites <- as.data.frame(methylation450@rowRanges@elementMetadata)

column_mask <- ifelse(colnames(methylation_clinical) %in% c('treatments', 'primary_site', 'disease_type','sites_of_involvement'), F, T)

methylation_clinical <- methylation_clinical[,column_mask]
write.csv(methylation_clinical, 'brca_methylation_clinical.csv')


site_mask <- !grepl('-', cpg_sites$gene) & !grepl(';', cpg_sites$gene) & !is.na(cpg_sites$gene) & complete.cases(betas)
betas <- betas[site_mask,]
cpg_sites <- cpg_sites[site_mask,]

write.csv(betas, 'brca_methylation_betas.csv')
write.csv(cpg_sites, 'brca_cpg_sites.csv')
```

```{r}
methylation_clinical <- read.csv("/home1/heclaire/490_cluster/analysis_data/brca_methylation_clinical.csv", row.names = 1)
betas <- read.csv("/home1/heclaire/490_cluster/analysis_data/brca_methylation_betas.csv", row.names = 1)
cpg_sites <- read.csv("/home1/heclaire/490_cluster/analysis_data/brca_cpg_sites.csv", row.names = 1)
```


(1) Naive Differential Methylation
```{r}
#masking out NAs
na_mask <- !is.na(methylation_clinical$age_at_diagnosis)
methylation_clinical <- methylation_clinical[na_mask,]
betas_clean <- betas[,na_mask]

median <- median(methylation_clinical$age_at_diagnosis)

methylation_clinical$age_category <- ifelse(methylation_clinical$age_at_diagnosis >= median, "old", "young")

#fitting linear models using a "target value"
young_mask <- methylation_clinical$age_category == "young"

methylation_clinical$ages <- !young_mask

mval <- t(apply(betas_clean, 1, function(x) log2(x/(1-x)))) 
#mvalue is another statistic for methylation, centered at 0 and ranges from -1 to 1

design <- model.matrix(~ages, data = methylation_clinical)
fit <- lmFit(mval, design)
fit2 <- eBayes(fit)
```

```{r}
#Extracting model into dataframe
dat <- data.frame(foldchange = fit[["coefficients"]][,2], logPvalue =  -log10(p.adjust(fit2[["p.value"]][,2],method='BY')), geneName = cpg_sites$gene)
dat$threshold <- as.factor(abs(dat$foldchange) < 1)

#Visualization
cols <- c("TRUE" = "grey", "FALSE" = "blue")
ggplot(data=dat, aes(x=foldchange, y = logPvalue, color=threshold)) +
  geom_point(alpha=.2, size=0.6) +
  scale_colour_manual(values = cols) +
  geom_vline(xintercept = 1, colour="#990000", linetype="dashed") + 
  geom_vline(xintercept = - 1, colour="#990000", linetype="dashed") +
  geom_hline(yintercept = 2, colour = "#990000", linetype="dashed") +
  theme(legend.position="none") +
  xlab("Fold Change") +
  ylab("-log10 p value") +
  theme_bw() +
  theme(legend.position = "none")
```
Question 1:

The volcano plot shows that there are a few genes that are hypermethylated with a significant p value, suggesting potential involvement in gene silencing mechanisms that may be linked to metastasis status in TCGA BRCA patients. These hypermethylated genes could serve as biomarkers for distinguishing metastatic from non-metastatic patients. Moreover, according to the graph, none of the genes are hypomethylated. 

(2) Direct comparison of methylation status to transcriptional activity

#...INSERT DESeq2 Stuff here to generate 'results'...
```{r}
no_na_mask<-!is.na(rna_clinical$definition) & !is.na(rna_clinical$age_at_index)
clean_clinical<-rna_clinical[no_na_mask,]

no_solid_tissue_normal_mask<- ifelse(clean_clinical$sample_type=="Solid Tissue Normal",F,T)
clean_clinical<- clean_clinical[no_solid_tissue_normal_mask,]

clean_count<- rna_counts[ ,no_na_mask]
clean_count<- clean_count[ ,no_solid_tissue_normal_mask]

less_20_mask<-rowSums(clean_count)>=20
clean_count<-clean_count[less_20_mask,]
clean_genes<- rna_genes[less_20_mask,]


dds <- DESeqDataSetFromMatrix(countData = clean_count,
                              colData = clean_clinical,
                              design= ~age_at_index+definition)

dds_obj <- DESeq(dds)

resultsNames(dds_obj)

results <- results(dds_obj, format = "DataFrame", contrast = c("definition","Primary solid Tumor","Metastatic"))

results <- data.frame(results)
write.csv(results, 'methylation_DEseq_results.csv')
```

```{r}
#you can also try looking at "upregulated" or "hypermethylated" !
upregulated <- results[(results$log2FoldChange > 1), 'gene_name']
hypermethylated <- dat[dat$foldchange > 1, 'geneName']
interest_genes <- intersect(upregulated, hypermethylated)
interest_genes
```
Question 3: 
WNT2 Gene: 
According to the UCSC genome browser, there are cpg sites at the promoter region for the WNT2 gene. Based on my analysis, WNT2 seems to be upregulated and hypermethylated in primary tumors compared to metastatic tumors, as shown by the positive log2FoldChange in the DESeq2 results. This is consistent with the results in the volcano plot. Since WNT2 has CpG sites in its promoter, this methylation might be silencing the gene in primary tumors. This  matches previous research "Frequent up-regulation of WNT2 in primary gastric cancer and colorectal cancer" that shows WNT2 can promote cancer cell invasion and migration, particularly in gastric and colorectal cancer. 
https://www.spandidos-publications.com/ijo/19/5/1003

LRRN4 Gene: 
According to this article "Prognostic value and biological function of LRRN4 in colorectal cancer", LRRN4 has been shown to promote cancer cell proliferation, migration, and survival in colorectal cancer (CRC) through pathways like RAS/MAPK and PI3K.  However, according to the UCSC Genome Browser, there are no cpg sites at the promoter region for the LRRN4 gene, making it unlikely that DNA methylation is responsible for regulating its expression. In the context of metastatic vs. non-metastatic TCGA BRCA patients, the difference in expression might be driven by other regulatory mechanisms, such as transcription factors or post-translational modifications, rather than methylation. 
https://link.springer.com/article/10.1186/s12935-022-02579-x

SLC35F1 Gene: 
Based on the DESeq2 analysis, the gene SLC35F1 appears to be hypermethylated in primary tumors compared to metastatic tumors, as indicated by its positive log2FoldChange value. Since the UCSC Genome Browser confirms the presence of cpg sites at the promoter region of SLC35F1, this hypermethylation could suggest potential gene silencing in primary tumors. Hypermethylation at promoter cpg sites is commonly associated with reduced gene expression, which could imply that SLC35F1 is downregulated in primary tumors. This is supported by the article "Aberrantly Methylated and Expressed Genes as Prognostic Epigenetic Biomarkers for Colon Cancer," as SLC35F1 was highlighted as one of the genes with significant methylation and expression differences in colon cancer tissues. This suggests its potential role in cancer progression or development. Therefore, the reduced methylation in metastatic tumors may suggest that SLC35F1 becomes reactivated or more highly expressed in metastatic cases. 

https://www.liebertpub.com/doi/10.1089/dna.2020.5591


(Extra) Making Boxplots
```{r}
GENE<-"SCTR"

gene_counts_mask <- rna_genes$gene_name == GENE
gene_betas_mask <- cpg_sites$gene == GENE

rna_clinical_tumor <- rna_clinical$definition == "Primary solid Tumor"
methylation_clinical_tumor <- methylation_clinical$definition == "Primary solid Tumor"

rna_clinical_normal <- rna_clinical$definition == "Solid Tissue Normal"
methylation_clinical_normal <- methylation_clinical$definition == "Solid Tissue Normal"

rna_tumor <- as.numeric(rna_counts[gene_counts_mask, rna_clinical_tumor])
methylation_tumor <- (betas[gene_betas_mask, methylation_clinical_tumor])

rna_normal <- as.numeric(rna_counts[gene_counts_mask, rna_clinical_normal])
methylation_normal <- (betas[gene_betas_mask, methylation_clinical_normal])
```

```{r}
boxplot(rna_normal, rna_tumor, xlab='Group', ylab='Counts', names=c('Normal', 'Tumor'))
```
